docs/architecture.md

```diff
@@ -0,0 +1,113 @@
+# Studio Architecture
+
+*N.B.* This is an aspirational document, and describes how parts of Studio currently are, and the basic framework that has been put in place to facilitate that. It does not describe everything about the current Studio code, its structure, data architecture or anything else.
+
+## Frontend Architecture
+
+Studio is divided into multiple single page applications - each one has a distinct function and purpose. It is plausible that one day, this will be collapsed into one SPA to rule them all, but today is not that day.
+
+*The main SPAs are:*
+Channel List - this is responsible for showing over all listings of channels under different categories (public, editable, etc.) and also for creating new channels, and creating and editing collections of channels.
+
+Channel Edit - this is responsible for showing the contents of an individual channel and allowing a user with the appropriate permissions to edit the contents and structure of that channel, it is the primary workhorse of the Studio user interface.
+
+Administration - this is an admin only page that allows people with staff privileges to see extensive information about user accounts, user created channels, and other detailed information in order to support users in using Studio.
+
+Settings - this allows a user to modify their user account settings, this could potentially be included in the other SPAs rather than having its own SPA.
+
+Each SPA has its own dedicated Django template that sets up the environment for the SPA. The templates for these can be found in `contentcuration/contentcuration/templates/`.
+
+## Where does the frontend code live?
+
+All code for SPAs should live in `contentcuration/contentcuration/frontend/<spaName>` e.g. the Channel List code lives in [`contentcuration/contentcuration/frontend/channelList`](../contentcuration/contentcuration/frontend/channelList).
+
+Code that is common across multiple SPAs is in the [`contentcuration/contentcuration/frontend/shared`](../contentcuration/contentcuration/frontend/shared) folder - this can be referenced with the [Webpack](https://webpack.js.org/concepts/) alias `shared/...` in ES6 imports.
```

Comment:
One thing I've occasionally seen is to prefix aliases with a `~` or `@` to differentiate them from normal paths. For example, the alias could be 

```js
      alias: {
        // explicit alias definitions (rather than modules) for speed
        '~edit_channel': path.resolve(staticJsDir, 'edit_channel'),
        '~less': path.resolve(staticJsDir, 'less'),
        '~utils': path.resolve(staticJsDir, 'utils'),
        '~shared': path.resolve(srcDir, 'shared'),
        '~frontend': srcDir,
        '~jquery': studioJqueryDir,
        // TODO just use modules alias
        '~rawJquery': jqueryDir,
      },
```